sklearn/exceptions.py

```diff
@@ -111,6 +111,28 @@ class EfficiencyWarning(UserWarning):
     to some reason which may be included as a part of the warning message.
     This may be subclassed into a more specific Warning class.
 
+    Examples
+    --------
+    >>> from sklearn.neighbors import NearestNeighbors
+    >>> from sklearn.exceptions import EfficiencyWarning
+    >>> from scipy.sparse import csr_matrix
+    >>> import warnings
+    >>> warnings.simplefilter('always', EfficiencyWarning)
+    >>> samples = [[0., 0., 0.], [0., .5, 0.], [1., 1., .5]]
+    >>> neigh = NearestNeighbors(n_neighbors=1)
+    >>> neigh.fit(samples)
+    NearestNeighbors(n_neighbors=1)
+    >>> X = csr_matrix([[0., 2., 0.], [1., 0., 1.], [3., 0., 1.]])
+    >>> neigh.effective_metric_ = 'precomputed'
```

Comment:
I would prefer not to adjust the learned parameter here. The following would also raise the warning:

```py
X = csr_matrix([[0., 2., 0.], [1., 0., 1.], [3., 0., 1.]])
neigh = NearestNeighbors(n_neighbors=1, metric='precomputed')
neigh.fit(X)
```